async_service/trio.py

```diff
@@ -161,16 +201,44 @@ def cancel(self) -> None:
     async def wait_finished(self) -> None:
         await self._finished.wait()
 
+    def _track_current_task(
+        self, parent: trio.hazmat.Task = None
+    ) -> Tuple[trio.Event, trio.CancelScope]:
+        current_task = trio.hazmat.current_task()
+        if parent is not None:
+            # This covers the case for the *root* task which doesn't have a
+            # parent.
+            self._service_task_dag[parent].append(current_task)
```

Comment:
No, I'd rather this raise an exception since at this stage (from within the runtime of the task) we should have the parent already tracked so this should be an exceptional case.

Reply:
I was thinking that in some cases another service could call a method on this one directly, but I guess that's not supposed to happen as they should use that external_service API? 

This will be annoying for tests that directly call service methods that in turn call `run_task()`, though. Any suggestions on how to fix those tests? We already have a few in trinity

Reply:
I'm still a little hesitant about this so I'm going to open an issue to track it so we can make the decision outside of the context of this PR

Reply:
I've fixed this in #37